openpype/hosts/max/plugins/publish/validate_loaded_plugin.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""Validator for Loaded Plugin."""
+import os
+from pyblish.api import InstancePlugin, ValidatorOrder
+from pymxs import runtime as rt
+
+from openpype.pipeline.publish import (
+    RepairAction,
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+from openpype.hosts.max.api.lib import get_plugins
+
+
+class ValidateLoadedPlugin(OptionalPyblishPluginMixin,
+                           InstancePlugin):
+    """Validates if the specific plugin is loaded in 3ds max.
+    Studio Admin(s) can add the plugins they want to check in validation
+    via studio defined project settings
+    If families = ["*"], all the required plugins would be validated
+    If families
+
+    """
+
+    order = ValidatorOrder
+    hosts = ["max"]
+    label = "Validate Loaded Plugins"
+    optional = True
+    actions = [RepairAction]
+
+    def get_invalid(self, instance):
+        """Plugin entry point."""
+        if not self.is_active(instance.data):
+            self.log.debug("Skipping Validate Loaded Plugin...")
+            return
+
+        required_plugins = (
+            instance.context.data["project_settings"]["max"]["publish"]
+                                 ["ValidateLoadedPlugin"]
+                                 ["family_plugins_mapping"]
+        )
+
+        if not required_plugins:
+            return
+
+        invalid = []
+
+        # get all DLL loaded plugins in Max and their plugin index
+        available_plugins = {
+            plugin_name.lower(): index for index, plugin_name in enumerate(
+                get_plugins())
+        }
+
+        for families, plugin in required_plugins.items():
+            # Out of for loop build the instance family lookup
+            instance_families = {instance.data["family"]}
+            instance_families.update(instance.data.get("families", []))
+            self.log.debug(f"{instance_families}")
+            # In the for loop check whether any family matches
+            match_families = {fam.strip() for fam in
+                              families.split(",") if fam.strip()}
+            self.log.debug(f"match_families: {match_families}")
+            has_match = "*" in match_families or match_families.intersection(
+                instance_families) or families == "_"
+            if not has_match:
+                continue
+
+            if not plugin:
+                return
+
+            plugin_name = plugin.format(**os.environ).lower()
```

Comment:
Why do we format against os environ? Should this get a comment as to why that's needed?

Reply:
it's for the {3DSMAX_VERSION} in application setting, for example, tyFlow has different version of the settings regarding to the max version i.e tyflow_2024.dlo and tyflow_2013.dlo.
3DSMAX_VERSION is already set in the env through Openpype plugins (it can be found when you type os.environ["3DSMAX_VERSION"] but it would be great we can figure out other ways to do it.

Reply:
Added comment would be nice.